ros2_batch_job/__main__.py

```diff
@@ -249,6 +258,9 @@ def get_args(sysargv=None):
     args = parser.parse_args(argv)
     args.build_args = build_args
     args.test_args = test_args
+    for name in ('sourcespace', 'buildspace', 'installspace'):
+        if name in args.white_space_in and getattr(args, name) != parser.get_default(name):
```

Comment:
Technically this isn't a sufficient check if the user passed an argument explicitly. It could very well be passed explicitly but have the same value as the default.

That being said I think it is good enough for this use case.

Reply:
Fair point. I could use `default=argparse.SUPPRESS` to catch when it is or isn't provided, but then we would be going back to adding fields to args after its been parsed and putting the defaults outside of the argument parser. This was the best balance between expected behavior and maintainability that I could think of.

Reply:
Just for context: this is what I have done in colcon for the exact same problem: colcon/colcon-core#288

But as I said this can stay as is.